base/bucket.go

```diff
@@ -561,3 +595,35 @@ func IsCasMismatch(bucket Bucket, err error) bool {
 
 	return false
 }
+
+func GetMaxVbnoForBucket(bucket Bucket) uint16 {
+
+	// TODO: Move to sg-bucket interface (https://github.com/couchbase/sync_gateway/issues/2418)
+
+	var maxVbNo uint16 = 1024
+	var err error
+
+	switch bucket := bucket.(type) {
+	case CouchbaseBucketGoCB:
+		maxVbNo, err = bucket.GetMaxVbno()
+	case *CouchbaseBucketGoCB:
+		maxVbNo, err = bucket.GetMaxVbno()
+	case CouchbaseBucketGoCBGoCouchbaseHybrid:
+		maxVbNo, err = bucket.GetMaxVbno()
+	case *CouchbaseBucketGoCBGoCouchbaseHybrid:
+		maxVbNo, err = bucket.GetMaxVbno()
+	case CouchbaseBucket:
+		maxVbNo, err = bucket.GetMaxVbno()
+	case *CouchbaseBucket:
+		maxVbNo, err = bucket.GetMaxVbno()
+	default:
+		Warn("Not a couchbase bucket - assuming walrus, setting maxVbNo=1024")
+	}
```

Comment:
Feels weird to have a switch that's handling 6 scenarios and making the same call for each one.  Does having multiple options in a single case work, or does that break the usage of bucket inside the case?

I realize this will be getting moved to the sg-bucket interface anyway, but could clean up, in case that work somehow gets deferred.